packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -21,6 +21,24 @@ import {
 import S3 from "aws-sdk/clients/s3";
 import { performance } from "perf_hooks";
 
+const parseCookieValue = (cookie: string) => {
+  return cookie.split(";").reduce<{ [key: string]: string }>((acc, curr) => {
+    const [key, value] = curr.split("=");
+    acc[key.trim()] = value.trim();
+    return acc;
+  }, {});
+};
+
+const checkPreviewRequest = (request: CloudFrontRequest) => {
+  const targetCookie = request.headers.cookie || [];
+  return targetCookie.some((cookieObj) => {
+    const cookieValue = parseCookieValue(cookieObj.value);
+    return (
+      cookieValue["__next_preview_data"] && cookieValue["__prerender_bypass"]
```

Comment:
We'll need to also validate that the signature of the _next_preview_data is correct using json web tokens. [Here is](https://github.com/vercel/next.js/blob/42e8dcf281526b5dde27448010e96edcd5ceb196/packages/next/next-server/server/api-utils.ts#L374) next-server implementation of it.

Reply:
Any guidance on how we will include these extra dependencies? For the cookie parser it's not a big deal because it doesn't have dependencies of its own, so we can do the same thing that is being done for `next-aws-cloudfront` and copy it to the lambda bundle.

For `jsonwebtoken`, that won't be possible (without copying over everything in `node_modules`). We could inline it by copying over the contents from [here](https://github.com/vercel/next.js/blob/42e8dcf281526b5dde27448010e96edcd5ceb196/packages/next/compiled/jsonwebtoken/index.js), but not sure if that's what you had in mind.

Reply:
Yeah I was going to suggest a bundler. Last time I checked `rollup` was the best option for libraries.

Reply:
I'll proceed with rollup, it's a stalwart of the bundlers.